api/query_engine.go

```diff
@@ -3,13 +3,22 @@ package api
 import (
 	"fmt"
 	"sort"
+	"time"
 
 	"github.com/raintank/metrictank/api/models"
 	"github.com/raintank/metrictank/mdata"
+	"github.com/raintank/metrictank/stats"
 	"github.com/raintank/metrictank/util"
 	"github.com/raintank/worldping-api/pkg/log"
 )
 
+var (
+	dpAdjGranularity  = stats.NewGauge32("api.dp_adj_granularity")
+	dpMaxGranularity  = stats.NewGauge32("api.dp_max_granularity")
+	aggAdjGranularity = stats.NewGauge32("api.agg_adj_granularity")
+	aggMaxGranularity = stats.NewGauge32("api.agg_max_granularity")
+)
```

Comment:
gauges just report their current value at each flush interval (in our case every 10 seconds). thus using them for this use case would only give us insights in a small fraction of the actual requests. I think a Meter32 is a better choice here, then we get aggregate statistics (min/mean/max) and potentially histograms once grafana adds support for heatmaps.  I think the names can also be clearer. (what does `adj` mean? `agg` is too vague, `archive` would be clearer, etc).  I'll rework this PR to make the needed changes.